ui/component/viewers/videoViewer/internal/plugins/videojs-hls-quality-selector/plugin.js

```diff
@@ -238,35 +287,36 @@ class HlsQualitySelectorPlugin {
    * @param {number} height - A number representing HLS playlist.
    */
   setQuality(height) {
+    const { doSetDefaultVideoQuality } = this.config;
     const qualityList = this.player.qualityLevels();
 
     // Set quality on plugin
     this._currentQuality = height;
+    doSetDefaultVideoQuality(height);
 
     if (this.config.displayCurrentQuality) {
-      this.setButtonInnerText(height === 'auto' ? 'auto' : height === 'original' ? 'original' : `${height}p`);
+      this.setButtonInnerText(
+        height === QUALITY_OPTIONS.AUTO
+          ? QUALITY_OPTIONS.AUTO
+          : height === QUALITY_OPTIONS.ORIGINAL
+          ? QUALITY_OPTIONS.ORIGINAL
+          : `${height}p`
+      );
     }
 
     for (let i = 0; i < qualityList.length; ++i) {
       const quality = qualityList[i];
-      quality.enabled = quality.height === height || height === 'auto' || height === 'original';
+      quality.enabled =
+        quality.height === height || height === QUALITY_OPTIONS.AUTO || height === QUALITY_OPTIONS.ORIGINAL;
     }
 
-    if (height === 'original') {
+    if (height === QUALITY_OPTIONS.ORIGINAL) {
       if (this.player.currentSrc() !== this.player.claimSrcOriginal.src) {
-        setTimeout(() => this.swapSrcTo('original'));
+        setTimeout(() => this.swapSrcTo(QUALITY_OPTIONS.ORIGINAL));
       }
     } else {
       if (!this.player.isLivestream && this.player.currentSrc() !== this.player.claimSrcVhs.src) {
         setTimeout(() => this.swapSrcTo('vhs'));
-
-        if (height !== 'auto') {
-          // -- Re-select quality --
-          // Until we have "persistent quality" implemented, we need to do this
-          // because the VHS internals default to "auto" when initialized,
-          // causing a GUI mismatch.
-          setTimeout(() => this.setQuality(height), 1000);
-        }
```

Comment:
## Switching from _Original_ to _vhs_ ends up in Auto
1. http://localhost:9090/@eevblog:7/eevblog-1466-stanford-solar-power-at:0
2. Select _Original_.
3. Select _144p_ --> it ends up in _Auto_ instead.  The network log is also showing it is fetching 360p.